driver-core/src/main/com/mongodb/internal/connection/LoadBalancedServer.java

```diff
@@ -0,0 +1,163 @@
+import com.mongodb.connection.ServerId;
+import com.mongodb.connection.ServerType;
+import com.mongodb.event.ServerClosedEvent;
+import com.mongodb.event.ServerDescriptionChangedEvent;
+import com.mongodb.event.ServerListener;
+import com.mongodb.event.ServerOpeningEvent;
+import com.mongodb.internal.async.SingleResultCallback;
+import com.mongodb.internal.session.SessionContext;
+import com.mongodb.lang.Nullable;
+import org.bson.types.ObjectId;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.mongodb.assertions.Assertions.isTrue;
+import static com.mongodb.connection.ServerConnectionState.CONNECTING;
+import static com.mongodb.internal.connection.ClusterableServer.ConnectionState.AFTER_HANDSHAKE;
+import static com.mongodb.internal.connection.ClusterableServer.ConnectionState.BEFORE_HANDSHAKE;
+import static java.util.Arrays.asList;
+
+public class LoadBalancedServer implements ClusterableServer {
+    private static final List<Integer> SHUTDOWN_CODES = asList(91, 11600);
+    private final AtomicBoolean closed = new AtomicBoolean();
+    private final ServerId serverId;
+    private final ConnectionPool connectionPool;
+    private final ConnectionFactory connectionFactory;
+    private final ServerListener serverListener;
+    private final ClusterClock clusterClock;
+
+    public LoadBalancedServer(final ServerId serverId, final ConnectionPool connectionPool, final ConnectionFactory connectionFactory,
+                              final ServerListener serverListener, final ClusterClock clusterClock) {
+        this.serverId = serverId;
+        this.connectionPool = connectionPool;
+        this.connectionFactory = connectionFactory;
+        this.serverListener = serverListener;
+        this.clusterClock = clusterClock;
+
+        serverListener.serverOpening(new ServerOpeningEvent(serverId));
+        serverListener.serverDescriptionChanged(new ServerDescriptionChangedEvent(serverId,
+                ServerDescription.builder()
+                        .ok(true)
+                        .state(ServerConnectionState.CONNECTED)
+                        .type(ServerType.LOAD_BALANCER)
+                        .address(serverId.getAddress())
+                        .build(),
+                ServerDescription.builder().address(serverId.getAddress()).state(CONNECTING).build()));
+    }
+
+    @Override
+    public void resetToConnecting() {
+        // no op
+    }
+
+    @Override
+    public void invalidate() {
+        // no op
+    }
+
+    @Override
+    public void invalidate(final ConnectionState connectionState, final Throwable reason, final int connectionGeneration,
+                           final int maxWireVersion) {
+        invalidate(connectionState, reason, null);
+    }
+
+
+    private void invalidate(final ConnectionState connectionState, final Throwable t, @Nullable final ObjectId serviceId) {
+        if (!isClosed()) {
+            if (t instanceof MongoSocketException
```

Comment:
Invalidate the pool only by serviceId.  

Note: there is no other SDAM error handling, since SDAM doesn't happen in LB mode